### PR TITLE
chore(django): remove some appsec code from django integration

### PR DESCRIPTION
**Pull request by emmettbutler**
This change uses the Core API abstraction layer to remove some code related to Application Security Monitoring from the Django integration module, clarifying the separation of concerns and ownership between the two areas.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by ZStriker19 — APPROVED** [review]

**Review by christophe-papazian — APPROVED** [review]

File: ddtrace/contrib/django/patch.py [hunk 1]
```diff
@@ -713,49 +708,17 @@ def traced_login(django, pin, func, instance, args, kwargs):
         request = get_argument_value(args, kwargs, 0, "request")
         user = get_argument_value(args, kwargs, 1, "user")
 
-        if not config._appsec_enabled or mode == "disabled":
+        if mode == "disabled":
```
Comment (ZStriker19) [inline ddtrace/contrib/django/patch.py:711]: wasn't there some confusing logic around when this should or should not be disabled? Are we changing that, or does this somehow just simplify it?

Comment (ZStriker19) [reply ddtrace/contrib/django/patch.py:711]: Oh you moved the appsec_enabled check into the login and auth methods that get called. Just curious, what's the reason for doing that? Doesn't it make what's actually going to run a bit more obscure?

**Review by ZStriker19 — APPROVED** [review]
Just a question, but lgtm!

File: ddtrace/contrib/django/patch.py (hunk 1 above)
Comment (emmettbutler) [reply ddtrace/contrib/django/patch.py:711]: My intention with that change is to remove references to the AppSec product from the contrib code. I'm still getting a sense for exactly how much of that approach is constructive.

**Pull request merged**